src/core/ddsi/src/ddsi_discovery_endpoint.c

```diff
@@ -160,10 +160,14 @@ static int sedp_write_endpoint_impl
   else
   {
     assert (xqos != NULL);
-    ps.present |= PP_PROTOCOL_VERSION | PP_VENDORID;
+    ps.present |= PP_PROTOCOL_VERSION | PP_VENDORID | PP_PRODUCT_VERSION;
```

Comment:
I wouldn't add it to endpoint discovery: the version is constant for all endpoints in a participant, endpoint discovery only runs after participant discovery and participant discovery messages are periodic. So putting in participant discovery both ensures you eventually will find out, and avoids sending it for potentially thousands of endpoints.